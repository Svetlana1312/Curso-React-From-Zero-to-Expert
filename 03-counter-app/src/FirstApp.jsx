import PropTypes from 'prop-types';


 const getMessage = () =>
 'Este es un mensaje en pantalla con anonima fuera del componente';

export const FirstApp = ({ title, subtitle, name}) => {
    //const newMessage = {
  //      message: 'Hola mundo',
   //     title: 'Svetlana'
   //  };
  // if (!title) 
   //     throw new Error('Title no esta establecido'); 
     
  return (
   <>
   <h1>{ title }</h1>
   <h2>{ subtitle }</h2>
   <p>{ name } </p>
   
   </> 
  )
}
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
   // title: 'Titulo por defecto',
    subtitle: 'Subtitulo por defecto',
    name: 'Nombre por defecto'
}
