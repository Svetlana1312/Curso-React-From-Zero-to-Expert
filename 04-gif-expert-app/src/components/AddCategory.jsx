import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
const [inputValue, setInputValue] = useState('');

const onInputChange = ({ target }) => {
 setInputValue(target.value);

}
const onSubmit = ( { target } )=>{
   event.preventDefault();
   const inputTrimmed = inputValue.trim();
   if(inputTrimmed.length<=1)
     return;
   //setCategories((categories) => [ inputValue, ...categories ]);
   onNewCategory( inputTrimmed );
   setInputValue('');
}
  return (
        <form onSubmit={ onSubmit } aria-label="form"> 
                <input 
                    type="text"
                    placeholder="Buscar gifs"
                    value={ inputValue }
                    onChange={ onInputChange }
                ></input>

        </form>
        
  )
}
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
