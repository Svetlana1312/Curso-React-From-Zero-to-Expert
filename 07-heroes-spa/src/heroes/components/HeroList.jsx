import { useMemo } from "react";
import { getHeroesByPublisher } from "../helpers"
import { HeroItem } from "./HeroItem";


export const HeroList = ({ publisher }) => {
    const heroes =  useMemo(() => getHeroesByPublisher( publisher ), [ publisher])


  return (
    <div className="row rows-cols-1 row-cols-md-3 g-3" >
        {
            heroes.map((heroe)=> (<HeroItem heroe={ heroe } key={ heroe.id }/>))
        }
    </div>
  )
}
