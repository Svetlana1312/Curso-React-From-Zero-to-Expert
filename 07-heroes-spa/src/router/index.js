export * from './PrivateRouter';
export * from './AppRouter'