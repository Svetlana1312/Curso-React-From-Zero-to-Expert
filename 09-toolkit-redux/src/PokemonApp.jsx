import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux"
import { getPokemons } from "./store/slices/pokemon";


export const PokemonApp = () => {

    const { isLoading, pokemons, page } = useSelector((state) => state.pokemons);
   
    const dispatch = useDispatch();
    useEffect(() => {
      dispatch (getPokemons(page));
    
      
    }, []);
    if( isLoading )
       return (<h1>Loading ....</h1> );
    
    return (
        <> 
                <h1>Pokemon App</h1>
                <hr/>
                <ul>
                    {
                        pokemons.map( pokemon => 
                            <li key={ pokemon.name }>{ pokemon.name }</li>
                    )}
                    
                </ul>
                <button onClick={ () => dispatch(getPokemons( page )) }>
                    Next
                </button>
        </>)
    
  
  
}
