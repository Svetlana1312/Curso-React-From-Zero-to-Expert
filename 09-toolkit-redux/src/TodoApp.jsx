
import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/api"


export const TodoApp = () => {
    // const {data: todos = [], isLoading } = useGetTodosQuery();
    const [todoId, setTodoId] = useState(1);
    const {data: todo = {}, isLoading } = useGetTodoQuery(todoId);
    const onPrevClick = ()=>{
        if(todoId === 1)
            return;
       setTodoId(todoId - 1);
    }
    const onNextClick = ()=>{
         setTodoId(todoId + 1);
    }
  return (
    <>
    <h1>Todos - RTK Query</h1>
    <hr />
    <h4>Loading ....: {isLoading ? 'True' : 'False' }</h4>
    <pre>
       { JSON.stringify(todo)
       }
    </pre>
    {/* <ul>
        {todos.map((todo)=><li key={todo.id}>
    <strong> {todo.completed?'DONE: ':'Pending: '}</strong>{todo.title}
        </li>)}
    </ul> */}

    <button onClick={ onPrevClick }>Previous</button>
    <button onClick={ onNextClick }>Next</button>
    </>
  )
}
