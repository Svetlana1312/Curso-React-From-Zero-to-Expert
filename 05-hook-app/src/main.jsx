import React from 'react'
import ReactDOM from 'react-dom/client'
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { Padre } from './07-tarea-memo/Padre'
import { TodoApp } from './08-useReducer/TodoApp'
import { MainApp } from './09-useContext/MainApp'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { SimpleForm } from './02-useEffect/SimpleForm'
//import { CounterApp } from './01-useState/CounterApp'
//import { HooksApp } from './HooksApp'
import './index.css'
//import './08-useReducer/intro-javascript'
import { BrowserRouter } from "react-router-dom";


ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
      <React.StrictMode>
        <MainApp />
   </React.StrictMode>
  
  </BrowserRouter>
 
)
