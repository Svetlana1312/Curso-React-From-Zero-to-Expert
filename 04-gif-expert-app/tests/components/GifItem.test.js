import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Pruebas de <GifItem />', () => { 
    const tituloPrueba = "Titulo de prueba";
    const urlPrueba = "https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=8f8dcbc2rk79w7lfibtxt2zjdy3jtxg9kn8bbo6vfq6ztf1r&rid=giphy.gif&ct=g";
   
    test('Debe coincidir con el shapshot', () => { 
        const {container}  = render(<GifItem title={ tituloPrueba }  url={ urlPrueba }/>);
        expect(container).toMatchSnapshot();
     });
     test('Debe mostrar el texto alternativo y la url indicados', () => { 
        render(<GifItem title={ tituloPrueba }  
                            url={ urlPrueba }/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(urlPrueba);
        expect(alt).toBe(tituloPrueba);
     });
     test('Debe mostrar el titulo indicado', () => { 
        render(<GifItem title={ tituloPrueba }  
                            url={ urlPrueba }/>);
        expect(screen.getByText(tituloPrueba)).toBeTruthy();
     });
 });