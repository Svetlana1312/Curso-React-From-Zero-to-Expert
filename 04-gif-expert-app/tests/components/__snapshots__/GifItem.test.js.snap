// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifItem /> Debe coincidir con el shapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de prueba"
      src="https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=8f8dcbc2rk79w7lfibtxt2zjdy3jtxg9kn8bbo6vfq6ztf1r&rid=giphy.gif&ct=g"
    />
    <p>
       
      Titulo de prueba
    </p>
  </div>
</div>
`;
